### ISSUE TITLE
Failed Security Check with Alexa

### ISSUE THREAD

**Issue by Preator** [mdhiggins/ESP8266-HTTP-IR-Blaster#76]
This is what I see when alexa goes to activate the esp8266: via Serial monitor - 

Connection received - MSG
Failed security check, signatures do not match
1: 4c0fed1451050166ec7388f486f5f4f9c183cf3528bb3d9ce9b6e0f7b3809057
2: a37ab9cf6ddb738299739d55e5c53b9e75a671a2fe3482202918c782a98696be

I thought this might have been addressed in yesterdays update but I recompiled and ran into the same problem. Any thoughts?

**Comment by mdhiggins** [comment]
I would double check your user_id setting. Usually when this fails it’s because it wasn’t exactly copied. Even 1 character off or a trailing space is enough to cause security check failure. The user_id is private so don’t post it here but if you want me to look you can PM me what user_id value you’re using

**Comment by mdhiggins** [comment]
Added some additional clarification to the readme as well as a serial output warning if the user_id isn't in the expected format and that authentication will likely fail

**Comment by mdhiggins** [comment]
Any progress?

**Comment by Preator** [comment]
I haven't had a chance to try your last push.  I had bypassed the sha check
to test the rest. But I am having trouble getting the ir to generate.
Trying to diagnose if it's a version issue with different node mcu... mine
is v0.9 which is ver 3 and yours is ver 2.  I will update you with my
progress.
Thanks for the follow up
Wes

**Comment by mdhiggins** [comment]
No need to ‘bypass’ the sha check, you can just not save the user_id with WiFiManager and the check won’t be performed.

I would definitely not bypass it though there’s no reason it shouldn’t be working. I suspect it may have been working but the issue is with the actual transmission of the IR commands
The V3 and V2 boards have the same pin assignments so they should be functionally equivalent

You can post your Serial output when trying to send commands and I’ll take a look
You can also post a picture of your build and I can just glance at it for any obvious wiring issues
Another debug step is to put a regular LED in place of the IR one and see if that lights up when sending commands or alternatively watch the IR LED using a camera that can pick up IR (some cell phones have IR filters though, the iPhone for example can only see IR with the front facing camera, not the rear, android varies)

**Comment by Preator** [comment]
ok, I have pushed your last code to the ESP8266 and still have no luck.  I
am diagnosing using a DS203 digital scope and a agilent U1241AN multimeter
and get no signal on any of the pins when i ask alexa to turn on the
device.  I have copied the serial output from boot through recognizing of
the alexa command.  it still doesn't recognize my amazon ID and the
formatting code you added say it should be in a format amznl.... but I can
not find any reference to an id that resembles that. Is it my developer
credentials or my regular amazon login ID? Thanks for any feedback!

...changed the ext IP, pass and port.  One Note had to change pin 10 to pin
12 as 10 was causing a hard reset and not a reboot. changed 12 to 15 but
not using that part of the code as far as I know.

ESP8266 IR Controller
Config pin GPIO12 set to: 1
mounted file system
reading config file
opened config file
{"hostname":"Office Air
Con","passcode":"***","port_str":"###","user_id":"","ip":"192.168.1.35","gw":"192.168.1.1","sn":"255.255.255.0"}
parsed json
*WM: Adding parameter
*WM: hostname
*WM: Adding parameter
*WM: passcode
*WM: Adding parameter
*WM: port_str
*WM: Adding parameter
*WM: user_id
Using Static IP
*WM:
*WM: AutoConnect
*WM: Connecting as wifi client...
*WM: Custom STA IP/GW/Subnet
*WM: 192.168.1.35
*WM: Using last saved values, should be faster
*WM: Connection result:
*WM: 3
*WM: IP Address:
*WM: 192.168.1.35
Default port changed
WiFi connected! User chose hostname 'Office Air Con' passcode '***' and
port '###'
mDNS started. Hostname is set to Office Air Con.local
Local IP: 192.168.1.35
URL to send commands: http://Office Air Con.local:###
HTTP Server started on port 697
Turning off the LED to save power.
External IP: XXXXXXXX
External IP address request took 3697 ms
Ready to send and receive IR signals
Connection received - MSG
Blasting off
Blocking incoming IR signals
B271368F:unknown:43
Transmission complete
Reenabling receiving
Turning off the LED to save power.



Happy ThanksGiving :)

Wes

**Comment by mdhiggins** [comment]
So it looks like you're trying to send an 'unknown' command which won't work
If the command is 'unknown' you need to send it as a RAW (see readme) or try scanning again with less interference to see if you can get the encoding to be read

What kind of remote/command are you trying to send?

**Comment by Preator** [comment]
I'm sending an on/off command from a carrier air conditioning remote. My
end goal is to be able to control the a.c. units in my house via alexa. I
live in Guam... need good a.c. control :) I didn't quite follow how to
issue the raw command also where is the file to alter the length of raw?
Sorry I'm a little new to arduino...
Thanks
Wes
P.S. feature request - add an air conditioning object on your alexa skill?

**Comment by Preator** [comment]
thanks for the updated readme file :) the sha security is fully functional.
The pins are working properly now that i know about unknown not actually
sending a command.  I used a sony remote and everything worked fine for the
tv.  When I went to working with the RAW json message though I don't think
I am capturing it correctly. I am not getting the RAW string is too long,
must edit somefile.h to correct error.  The device shows two signals though
when I aim the remote and switch it on or off. I tried adding the shorter
raw data to the longer one in the json line but no response from the ac
unit.  I have copied and pasted the esp receive lines and the json line i
am sending to the ac. is there anything obvious that you see wrong here?
Again thanks for the help,
Wes

...ESP receive screen

Time SentCommandTypeLengthAddress
03:02:53 <http://192.168.1.35:697/received?id=1> 8BC0374F UNKNOWN 7 0X
03:02:52 <http://192.168.1.35:697/received?id=2> 4B291791 UNKNOWN 34 0X

... lines sent through the skill for on and off

Power On
Power Off
